lib/schemas.js

```diff
@@ -0,0 +1,94 @@
+/* eslint-disable new-cap */
+'use strict';
+
+const {
+  DATA_UNIT_TYPE,
+
+  HANDSHAKE,
+  PARCEL_HEADER,
+  STREAM_HEADER,
+  CHUNK_HEADER,
+  CHUNK_PAYLOAD,
+  ACK,
+
+  DATA_UNIT_TYPES,
+  ENCODINGS,
+  COMPRESSIONS,
+  HANDSHAKE_STATUSES,
+} = require('./constants');
+
+const UInt = {
+  parse: (buffer, size, offset = 0) => buffer.readUIntLE(offset, size),
+};
+
+const Binary = {
+  parse: (buffer, size, offset = 0) => buffer.slice(offset, offset + size),
+};
+
+const Enum = values => ({
+  parse: (buffer, size, offset = 0) => values[buffer.readUIntLE(offset, size)],
+});
+
+const Struct = schemas => ({
+  parse: (buffer, size, offset = 0) => {
+    const obj = {};
+    for (const { name, size, type } of schemas) {
+      obj[name] = type.parse(buffer, size, offset);
+      offset += size;
+    }
+    return obj;
+  },
+});
+
+module.exports = [
+  {
+    name: DATA_UNIT_TYPE,
+    size: 1,
+    type: Enum(DATA_UNIT_TYPES),
+  },
+  {
+    name: HANDSHAKE,
+    size: 35,
```

Comment:
I am not sure about whether it's a good idea but maybe we can create a function `newStruct` counting the total size so instead of `{ name: ... , size: ..., type: ... }` there would be just `newStruct(HANDSHAKE, [{ ... }, { ... }, { ... }])`